manifests/init.pp

```diff
@@ -104,6 +111,9 @@
     }
   }
 
+  $caddy_dl_url = "${caddy_url}/${arch}?plugins=${caddy_features}&license=${caddy_license}&telemetry=${caddy_telemetry}"
```

Comment:
This should be further customized, so you can just grab it from your local repo, but that's beyond the scope of this.

Reply:
I think this will be no longer supported and the user has to build his own binary: 
* https://github.com/caddyserver/caddy/releases